### PR TITLE
feat: implement switch component

### PR DESCRIPTION
**Pull request by loido**
## Check List
- [ ] Your code builds clean without any errors or warnings.
- [ ] Check coding convention.
- [ ] Updated Storybook components and stories to reflect any changes made to the UI.
- [ ] Test covers all edge cases and coverage is greater than or equal to 80%.

## Proof of Completeness
Screenshot changes introduced" width="250" />
OR
Video link

### CODE REVIEW

File: src/components/Switch/Switch.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  TRACK_RADIUS,
+  TRACK_WIDTH,
+} from './switch.utils'
+
+const Switch: React.FC<SwitchProps> = ({
+  value,
+  onValueChange,
+  trackColor = defaultTheme.colors.textColor,
+  thumbSize = THUMB_SIZE,
+  thumbColor = 'white',
```
Comment (anhle10051996) [inline src/components/Switch/Switch.tsx:22]: could we also use color from theme ?

File: src/components/Switch/Switch.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            </Label>
+          )}
+        </TrackInActive>
+      </TrackContainer>
+      <Thumb testID="thumb" style={thumbAnimatedStyle} {...{thumbSize, thumbColor}} />
+    </SwitchContainer>
+  )
+}
+
+const SwitchContainer = styled.Pressable<ISwitchContainer>(({trackSize, trackMargin}) => ({
```
Comment (anhle10051996) [inline src/components/Switch/Switch.tsx:104]: why you not use styled(Pressable)

File: src/components/Switch/Switch.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  paddingRight: trackMargin,
+}))
+
+const Thumb = styled(Animated.View)<IThumb>(({thumbSize}) => ({
+  width: thumbSize,
+  height: thumbSize,
+  borderRadius: thumbSize,
+}))
+
+const TrackContainer = styled.View<ITrack>(({trackSize}) => ({
```
Comment (anhle10051996) [inline src/components/Switch/Switch.tsx:118]: same as above

File: src/components/Switch/Switch.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  SwitchVariant,
+  TEXT_INSIDE,
+  TEXT_INSIDE_COLOR,
+  THUMB_SIZE,
+  TRACK_RADIUS,
+  TRACK_WIDTH,
+} from './switch.utils'
+
+const Switch: React.FC<SwitchProps> = ({
+  value,
```
Comment (anhle10051996) [inline src/components/Switch/Switch.tsx:18]: as description for props

File: src/components/Switch/Switch.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+import {
+  SPACING_INSIDE,
+  SPACING_OUTSIDE,
+  SwitchVariant,
+  TEXT_INSIDE,
+  TEXT_INSIDE_COLOR,
+  THUMB_SIZE,
+  TRACK_RADIUS,
+  TRACK_WIDTH,
+} from './switch.utils'
```
Comment (anhle10051996) [inline src/components/Switch/Switch.tsx:15]: I believe it is more likely the constants file , not the utils file.

**Comment by anhle10051996** [comment]
please provide your image or video of your component and test result

File: src/components/Switch/Switch.tsx (hunk 1 above)
Comment (hangnguyensaigontech) [reply src/components/Switch/Switch.tsx]: updated

File: src/components/Switch/Switch.tsx (hunk 2 above)
Comment (hangnguyensaigontech) [reply src/components/Switch/Switch.tsx]: updated

File: src/components/Switch/Switch.tsx (hunk 3 above)
Comment (hangnguyensaigontech) [reply src/components/Switch/Switch.tsx]: updated

File: src/components/Switch/Switch.tsx (hunk 5 above)
Comment (hangnguyensaigontech) [reply src/components/Switch/Switch.tsx]: updated

File: src/components/Switch/Switch.tsx (hunk 4 above)
Comment (hangnguyensaigontech) [reply src/components/Switch/Switch.tsx:112]: added

**Review by loc-nguyenthien — APPROVED** [review]

**Review by vunguyendhSTS — APPROVED** [review]

File: example/src/stories/Switch.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+import {View} from 'react-native'
+import {Switch} from 'rn-base-component'
+
+export default {
+  title: 'components/Switch',
+  component: Switch,
+} as ComponentMeta<typeof Switch>
+
+export const Basic: ComponentStory<typeof Switch> = args => {
```
Comment (honghoangsts) [inline example/src/stories/Switch.stories.tsx:13]: dont use `args` please use `rest`

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: example/src/stories/Switch.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+  return (
+    <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
+      <Switch
+        {...args}
+        variant="outside"
+        value={isActive}
+        onValueChange={onValueChange}
+        thumbColor="green"
+        trackColor={{active: 'blue', inActive: 'gray'}}
```
Comment (honghoangsts) [inline example/src/stories/Switch.stories.tsx:28]: don't use magic string

**Review by chungnguyenhoang — APPROVED** [review]
